Libplanet.Net/Consensus/ContextTimeoutOption.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+
+namespace Libplanet.Net.Consensus
+{
+    /// <summary>
+    /// A options class to configure <see cref="Context{T}"/> timeout for each <see cref="Step"/>.
+    /// </summary>
+    public class ContextTimeoutOption
+    {
+        public ContextTimeoutOption(
+            int proposeSecondBase = 5,
+            int preVoteSecondBase = 5,
+            int preCommitSecondBase = 5,
+            int proposeMultiplier = 1,
+            int preVoteMultiplier = 1,
+            int preCommitMultiplier = 1)
+        {
+            if (proposeSecondBase <= 0)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(proposeSecondBase),
+                    "ProposeSecondBase must be greater than 0.");
+            }
+
+            ProposeSecondBase = proposeSecondBase;
+
+            if (preVoteSecondBase <= 0)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(preVoteSecondBase),
+                    "PreVoteSecondBase must be greater than 0.");
+            }
+
+            PreVoteSecondBase = preVoteSecondBase;
+
+            if (preCommitSecondBase <= 0)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(preCommitSecondBase),
+                    "PreCommitSecondBase must be greater than 0.");
+            }
+
+            PreCommitSecondBase = preCommitSecondBase;
+
+            if (proposeMultiplier <= 0)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(proposeMultiplier),
+                    "ProposeMultiplier must be greater than 0.");
+            }
+
+            ProposeMultiplier = proposeMultiplier;
+
+            if (preVoteMultiplier <= 0)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(preVoteMultiplier),
+                    "PreVoteMultiplier must be greater than 0.");
+            }
+
+            PreVoteMultiplier = preVoteMultiplier;
+
+            if (preCommitMultiplier <= 0)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(preCommitMultiplier),
+                    "PreCommitMultiplier must be greater than 0.");
+            }
+
+            PreCommitMultiplier = preCommitMultiplier;
+        }
+
+        // FIXME: This is a prevention of initialization with default. the fields can be initialized
+        // with value zero.
+        // ReSharper disable MemberInitializerValueIgnored
+        public int ProposeSecondBase { get; } = 5;
+
+        public int PreVoteSecondBase { get; } = 5;
+
+        public int PreCommitSecondBase { get; } = 5;
+
+        public int ProposeMultiplier { get; } = 1;
+
+        public int PreVoteMultiplier { get; } = 1;
+
+        public int PreCommitMultiplier { get; } = 1;
```

Comment:
I think you can remove default assignments here as parameter-less default constructor is not available. 🤔

Reply:
I see. One thing I noticed is that struct default sets all values to zero, but class default treats the same as null.
https://github.com/planetarium/libplanet/pull/2382/commits/7fa12b6a7634b75b2dae35b83a303e094d1b0cf7#diff-95065cc3ba907122e5352fa2ee987c580d32bfe93d8c8e5a71b70a23e2767aeb